File: Domain/DTOs/ChildCardDto.cs
using WebApi.DTOs.Base;

namespace Domain.DTOs
{
    public class ChildCardDto : BaseDto
    {
        public long ContactNumbersID { get; set; }

        public String? BirthOfDate { get; set; }
        public DateTime? CardFillDate { get; set; }

        public long? NumberOfBrothers { get; set; }
        public long? NumberOfSisters { get; set; }
        public long? ChildArrangement { get; set; }
        public long? NumberOfSpicialNeeds { get; set; }
        public long? UserID { get; set; }

        public string? KindergartenName { get; set; }
        public string? StudentName { get; set; }
        public string? FatherName { get; set; }
        public string? GrandFatherName { get; set; }
        public string? FamilyName { get; set; }
        public string? BirthPlace { get; set; }
        public string? Nationality { get; set; }
        public string? Religion { get; set; }
        public string? Address { get; set; }
        public string? FatherProfession { get; set; }
        public string? FatherWorkPhone { get; set; }
        public string? phoneNumber { get; set; }
        public string? FatherWorkPlace { get; set; }
        public string? MotherWorkPhone { get; set; }
        public string? MotherWorkPlace { get; set; }
        public string? FatherEducationLevel { get; set; }
        public string? MotherEducationLevel { get; set; }
        public string? EmergencyPhoneNumber { get; set; }
        public string? childLivesWith { get; set; }
        public string? ParentsSituation { get; set; }
        public string? DiseasesOfTheChild { get; set; }
        public string? TheAuthoritySupervisingDisability { get; set; }
        public string? ExtraInfo { get; set; }
        public string? GurdianName { get; set; }
        public string? TeacherName { get; set; }
        public string? ManegerName { get; set; }
        public string? TheTypeOfConditionTheChildIsFacing { get; set; }
        public string? ConditionDescription { get; set; }
        public string? ConditionReasons { get; set; }
        public string? ConditionRelatedIndicators { get; set; }
        public string? HowTheCaseMonitoredDone { get; set; }
        public string? RejectReason { get; set; }

        public bool? DidStudentHaveSurgery { get; set; }
        public bool? DidHeTakesMedication { get; set; }
        public bool? DidHeTakesHealthVaccinations { get; set; }
        public bool? ChildReceivedAllTheRequiredVaccinations { get; set; }
        public bool? FromSpecialNeeds { get; set; }
        public bool? IsThereAnyoneInTheFamilyWithSpecialNeeds { get; set; }
        public bool? HaveChaildAttendedKindergartenBefore { get; set; }
        public bool? isCheck { get; set; } = false;
        public bool? Status { get; set; }


        public long? PersonalImageID { get; set; }
        public long? BirthCertificateImageID { get; set; }
        public long? ParentIdentifyImageID { get; set; }
        public long? StudentPersonImageID { get; set; }
    }
}

File: frontend/lib/model/childCardDto.dart
class ChildCardDto {
  num? id;
  String? studentName;
  String? fatherName;
  String? grandFatherName;
  String? familyName;
  String? birthOfDate;
  String? address;
  String? phoneNumber;
  String? childLivesWith;
  String? parentsSituation;
  num? userID;
  bool? isCheck;
  bool? status;
  String? rejectReason;

  ChildCardDto({
    this.id,
    this.studentName,
    this.fatherName,
    this.grandFatherName,
    this.familyName,
    this.birthOfDate,
    this.address,
    this.phoneNumber,
    this.childLivesWith,
    this.parentsSituation,
    this.userID,
    this.isCheck,
    this.status,
    this.rejectReason,
  });

  ChildCardDto.fromJson(Map<String, dynamic> json) {
    if (json.containsKey('id')) {
      id = json['id'];
    } else {
      id = 0;
    }
    studentName = json['studentName'];
    fatherName = json['fatherName'];
    grandFatherName = json['grandFatherName'];
    familyName = json['familyName'];
    birthOfDate = json['birthOfDate'];
    address = json['address'];
    phoneNumber = json['phoneNumber'];
    childLivesWith = json['childLivesWith'];
    parentsSituation = json['parentsSituation'];
    userID = json['userID'];
    isCheck = json['isCheck'];
    status = json['status'];
    rejectReason = json['rejectReason'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['studentName'] = studentName;
    data['fatherName'] = fatherName;
    data['grandFatherName'] = grandFatherName;
    data['familyName'] = familyName;
    data['birthOfDate'] = birthOfDate;
    data['address'] = address;
    data['phoneNumber'] = phoneNumber;
    data['childLivesWith'] = childLivesWith;
    data['parentsSituation'] = parentsSituation;
    data['id'] = id;
    data['userID'] = userID;
    data['isCheck'] = isCheck;
    data['status'] = status;
    data['rejectReason'] = rejectReason;
    return data;
  }
}

/////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////

class ChildCardPaginationResult {
  List<ChildCardDto>? data;
  num? count;

  ChildCardPaginationResult({this.data, this.count});

  ChildCardPaginationResult.fromJson(Map<String, dynamic> json) {
    if (json['data'] != null) {
      data = <ChildCardDto>[];
      json['data'].forEach((v) {
        data!.add(ChildCardDto.fromJson(v));
      });
    }
    count = json['count'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['childCard'] = this.data!.map((v) => v.toJson()).toList();
    data['count'] = count;
    return data;
  }
}
